### PR TITLE
Return mixed type if property fetch scope is null

### LINKED ISSUE #5980: "PHPStan\Reflection\ClassReflection::getProperty(): Argument #2 ($scope) must be of type PHPStan\Reflection\ClassMemberAccessAnswerer, null given
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.10.3              |
| Installed as   | composer dependency                          |

fatal error
```
"PHPStan\Reflection\ClassReflection::getProperty(): Argument #2     ($scope) must be of type PHPStan\Reflection\ClassMemberAccessAnswerer,  null given, called in vendor/rector/rector/packages/NodeTypeResolver/NodeTypeResolver/PropertyFetchTypeResolver.php:135". 
```

## Minimal PHP Code Causing Issue

Sorry, I couldn't find a code to reproduce the issue on https://getrector.org/demo, does it use the latest 0.10.3 version? (I got this fatal after updating from 0.10.1).

It failed on a trait that used Eloquent properties (but not only), example:

```php
trait UserHasAddress
{
    public function getAddressLine1(): ?string
    {
        return $this->address_line_1;
    }
}
```
But, that trait is pretty bit, and even the whole trait doesn't  cause any reports on https://getrector.org/demo using all rectors and sets we use on the project. I hope it will help somehow.

rectors:

```php
use Rector\CodeQuality\Rector\Assign\CombinedAssignRector;
use Rector\CodeQuality\Rector\Class_\CompleteDynamicPropertiesRector;
use Rector\CodeQuality\Rector\ClassMethod\DateTimeToDateTimeInterfaceRector;
use Rector\CodeQuality\Rector\For_\ForRepeatedCountToOwnVariableRector;
use Rector\CodeQuality\Rector\For_\ForToForeachRector;
use Rector\CodeQuality\Rector\FuncCall\CompactToVariablesRector;
use Rector\CodeQuality\Rector\If_\CombineIfRector;
use Rector\CodeQuality\Rector\If_\ConsecutiveNullCompareReturnsToNullCoalesceQueueRector;
use Rector\CodeQuality\Rector\If_\ExplicitBoolCompareRector;
use Rector\CodeQuality\Rector\Return_\SimplifyUselessVariableRector;
use Rector\CodeQuality\Rector\Ternary\ArrayKeyExistsTernaryThenValueToCoalescingRector;
use Rector\CodingStyle\Rector\Assign\SplitDoubleAssignRector;
use Rector\CodingStyle\Rector\ClassMethod\MakeInheritedMethodVisibilitySameAsParentRector;
use Rector\CodingStyle\Rector\FuncCall\CallUserFuncCallToVariadicRector;
use Rector\CodingStyle\Rector\FuncCall\ConsistentImplodeRector;
use Rector\CodingStyle\Rector\FuncCall\StrictArraySearchRector;
use Rector\CodingStyle\Rector\If_\NullableCompareToNullRector;
use Rector\CodingStyle\Rector\String_\SplitStringClassConstantToClassConstFetchRector;
use Rector\CodingStyle\Rector\String_\SymplifyQuoteEscapeRector;
use Rector\CodingStyle\Rector\Switch_\BinarySwitchToIfElseRector;
use Rector\Core\Configuration\Option;
use Rector\Core\ValueObject\PhpVersion;
use Rector\Php71\Rector\FuncCall\RemoveExtraParametersRector;
use Rector\Php74\Rector\Closure\ClosureToArrowFunctionRector;
use Rector\Php74\Rector\FuncCall\ArraySpreadInsteadOfArrayMergeRector;
use Rector\Php74\Rector\LNumber\AddLiteralSeparatorToNumberRector;
use Rector\Php74\Rector\Property\TypedPropertyRector;
use Rector\Privatization\Rector\Class_\FinalizeClassesWithoutChildrenRector;
use Rector\PSR4\Rector\FileWithoutNamespace\NormalizeNamespaceByPSR4ComposerAutoloadRector;
use Rector\Set\ValueObject\SetList;
use Rector\TypeDeclaration\Rector\Closure\AddClosureReturnTypeRector;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;


return static function (ContainerConfigurator $containerConfigurator): void {
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::PHP_VERSION_FEATURES, PhpVersion::PHP_80);

    $parameters->set(Option::SETS, [
        SetList::CODE_QUALITY,
        SetList::PHP_74,
        SetList::PHP_80,
    ]);

    $parameters->set(Option::SKIP, [
        Rector\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector::class,
        Rector\Php80\Rector\FunctionLike\UnionTypesRector::class,
        SymplifyQuoteEscapeRector::class,
        SimplifyUselessVariableRector::class,
        RemoveExtraParametersRector::class,
        ArraySpreadInsteadOfArrayMergeRector::class,
        AddLiteralSeparatorToNumberRector::class,
        TypedPropertyRector::class,
        NormalizeNamespaceByPSR4ComposerAutoloadRector::class,
        AddClosureReturnTypeRector::class,
        ClosureToArrowFunctionRector::class,
        FinalizeClassesWithoutChildrenRector::class,
        DateTimeToDateTimeInterfaceRector::class,
    ]);

    $services = $containerConfigurator->services();
    $services->set(ArrayKeyExistsTernaryThenValueToCoalescingRector::class);
    $services->set(CombineIfRector::class);
    $services->set(CombinedAssignRector::class);
    $services->set(CompactToVariablesRector::class);
    $services->set(CompleteDynamicPropertiesRector::class);
    $services->set(ConsecutiveNullCompareReturnsToNullCoalesceQueueRector::class);
    $services->set(ExplicitBoolCompareRector::class);
    $services->set(ForRepeatedCountToOwnVariableRector::class);
    $services->set(ForToForeachRector::class);
    $services->set(BinarySwitchToIfElseRector::class);
    $services->set(CallUserFuncCallToVariadicRector::class);
    $services->set(ConsistentImplodeRector::class);
    $services->set(MakeInheritedMethodVisibilitySameAsParentRector::class);
    $services->set(NullableCompareToNullRector::class);
    $services->set(SplitDoubleAssignRector::class);
    $services->set(SplitStringClassConstantToClassConstFetchRector::class);
    $services->set(StrictArraySearchRector::class);
    $services->set(FinalizeClassesWithoutChildrenRector::class);
};
```

## Expected Behaviour

Don’t have a fatal error :)

### PR DESCRIPTION
**Pull request by zingimmick**
Close #5980

### CODE REVIEW

**Comment by samsonasik** [comment]
@zingimmick could you add fixture test for it? Thank you.

**Comment by zingimmick** [comment]
I am not sure where I should add the fixture test, because I am not sure which is the responsible rule.

- It will throw a TypeError when you add to any rector's fixture test, such as `ArgumentAdderRector`.
- Can't reproduce it in the fixture test of `PropertyFetchTypeResolver`.

Here is the code, you can test it without install any dependencies.
```php
<?php

class Value
{
    public function doSomething()
    {

    }
}

class Property
{
    public $value;
}

trait InteractsWithProperty
{
    public function doSomething(Property $property)
    {
        $property->value->doSomething();
    }
}

final class SomeExampleClass
{
    use InteractsWithProperty;
}
```

**Comment by zingimmick** [comment]
@samsonasik

**Comment by zingimmick** [comment]
BTW, This is just the way to make it run, if there is a better way to solve this problem, feel free to close it.

**Comment by samsonasik** [comment]
You can add to `ArgumentAdderRector` fixture test then, just one of the rector rule to show it.

**Comment by zingimmick** [comment]
> You can add to `ArgumentAdderRector` fixture test then, just one of the rector rule to show it.

Already add to ArgumentAdderRector fixture test, see https://github.com/rectorphp/rector/pull/5998/checks?check_run_id=2202649797#step:5:58

**Comment by samsonasik** [comment]
@zingimmick thank you 👍

**Review by samsonasik — APPROVED** [review]

File: rules-tests/Arguments/Rector/ClassMethod/ArgumentAdderRector/Fixture/fixture4.php.inc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+}
+
+final class SomeExampleClass
+{
+    use InteractsWithProperty;
+}
+
+?>
+-----
```
Comment (samsonasik) [inline rules-tests/Arguments/Rector/ClassMethod/ArgumentAdderRector/Fixture/fixture4.php.inc:32]: this second part seems equal with above, you can remove it, and change fixture file with like `skip_mixed_scope.php.inc` and change class name to `class SkipMixedScope`

Comment (zingimmick) [reply rules-tests/Arguments/Rector/ClassMethod/ArgumentAdderRector/Fixture/fixture4.php.inc]: fixed 🎉

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**